qir/qsharp-target-tests/TargetTestScripts/__init__.py

```diff
@@ -1,63 +1,509 @@
 import os
+import shutil
+import tempfile
+
+import subprocess
+import pytest
+
+
 SCRIPT_DIR = os.path.dirname(__file__)
 
-available1 = [  # "Intrinsic/ApplyControlledX.qs",
-    # "Intrinsic/ApplyControlledZ.qs",
-    # "Intrinsic/ApplyUncontrolledH.qs",
-    # "Intrinsic/ApplyUncontrolledRx.qs",
-    # "Intrinsic/ApplyUncontrolledRy.qs",
-    # "Intrinsic/ApplyUncontrolledRz.qs",
-    # "Intrinsic/ApplyUncontrolledS.qs",
-    # "Intrinsic/ApplyUncontrolledSAdj.qs",
-    # "Intrinsic/ApplyUncontrolledT.qs",
-    # "Intrinsic/ApplyUncontrolledTAdj.qs",
-    # "Intrinsic/ApplyUncontrolledX.qs",
-    # "Intrinsic/ApplyUncontrolledY.qs",
-    # "Intrinsic/ApplyUncontrolledZ.qs",
-    # "Intrinsic/MZ.qs",
-    # "Intrinsic/Reset.qs",
-    "Decompositions/AssertOperationsEqualInPlace.qs",
-    "Decompositions/AssertOperationsEqualReferenced.qs",
-    "Decompositions/CCNOTFromCCZ.qs",
-    "Decompositions/CNOTFromSinglyControlled.qs",
+available1 = [
+    "Decompositions/CCNOT.qs",
+    "Decompositions/CCNOTAdjoint.qs",
+    "Decompositions/CCNOTControlled.qs",
+    "Decompositions/CCNOTAdjointControlled.qs",
+    "Decompositions/CCNOTControlledAdjoint.qs",
+    "Decompositions/CNOT.qs",
+    "Decompositions/CNOTAdjoint.qs",
+    "Decompositions/CNOTControlled.qs",
+    "Decompositions/CNOTAdjointControlled.qs",
+    "Decompositions/CNOTControlledAdjoint.qs",
     "Decompositions/CX.qs",
-    "Decompositions/CYFromCNOT.qs",
-    "Decompositions/CZFromSinglyControlled.qs",
-    "Decompositions/ExpFracFromExpUtil.qs",
-    "Decompositions/ExpFromExpUtil.qs",
-    "Decompositions/ExpUtil.qs",
+    "Decompositions/CXAdjoint.qs",
+    "Decompositions/CXControlled.qs",
+    "Decompositions/CXAdjointControlled.qs",
+    "Decompositions/CXControlledAdjoint.qs",
+    "Decompositions/CY.qs",
+    "Decompositions/CYAdjoint.qs",
+    "Decompositions/CYControlled.qs",
+    "Decompositions/CYAdjointControlled.qs",
+    "Decompositions/CYControlledAdjoint.qs",
+    "Decompositions/CZ.qs",
+    "Decompositions/CZAdjoint.qs",
+    "Decompositions/CZControlled.qs",
+    "Decompositions/CZAdjointControlled.qs",
+    "Decompositions/CZControlledAdjoint.qs",
+    # TODO(tfr): "Decompositions/ExpFrac.qs",
```

Comment:
How come these are a todo? They are not supported in any of the profiles, so I don't think there should be tests for them.

Reply:
OK, I will remove them.